src/detect-base64-decode.c

```diff
@@ -89,17 +89,18 @@ int DetectBase64DecodeDoMatch(DetectEngineThreadCtx *det_ctx, const Signature *s
 
     decode_len = MIN(payload_len, data->bytes);
 
+    DEBUG_VALIDATE_BUG_ON(decode_len < 0);
```

Comment:
Which could be negative ? payload_len or data->bytes ?

Reply:
`payload_len`, I believe as it is being calculated and subtracted from right above this. I tried to make this better in the following commit btw https://github.com/OISF/suricata/pull/11783/commits/ee6263eb328c017f7dc408b1d7957f0e460188b9 Lmk wdyt?

Reply:
Indeed https://github.com/OISF/suricata/commit/ee6263eb328c017f7dc408b1d7957f0e460188b9 looks good